Fixed undefined method error when doing http basic authentication.

victorteokw opened the pull request:
Fixed error `undefined method `split' for nil:NilClass` when

``` ruby
name, password = ActionController::HttpAuthentication::Basic.user_name_and_password(request)
```

when `request.authorization` returns nil, `auth_param` and `auth_scheme` raise error.

robertomiranda committed:
Remove attributes_protected_by_default reference, since MassAssignmentSecurity was removed from ActiveModel f8c9a4d3e88181

sgrif committed:
Time columns should support time zone aware attributes

The types that are affected by `time_zone_aware_attributes` (which is on
by default) have been made configurable, in case this is a breaking
change for existing applications.

skanev committed:
Add an `:if_exists` option to `drop_table`

If set to `if_exists: true`, it generates a statement like:

    DROP TABLE IF EXISTS posts

This syntax is supported in the popular SQL servers, that is (at least)
SQLite, PostgreSQL, MySQL, Oracle and MS SQL Sever.

Closes #16366.

morgoth committed:
Add ActiveModel::Errors#details

To be able to return type of validator, one can now call `details`
on Errors instance:

```ruby
class User < ActiveRecord::Base
  validates :name, presence: true
end
```

```ruby
user = User.new; user.valid?; user.errors.details
=> {name: [{error: :blank}]}
```

nobu committed:
Use IO::NULL always

tgxworld committed:
Use parallel bundler install.

Travis recommends a default of 3.
See: http://docs.travis-ci.com/user/languages/ruby/#Default-Test-Script

rafaelfranca committed:
Merge pull request #18410 from seuros/ripfilter

remove *_filter

caike committed:
Add _mailer suffix to i18n path

For mailers created via generators

Follow up to #18074

yoshiokatsuneo committed:
ActiveRecord: release connection on reconnect failure.

When trying to checkout connection from connection pool,
checkout()(and checkout_and_verify) verify whether the connection
is active or not.
And, if the connection is not active, connection adapters try to
reconnect to server.  And, if database is down at this moment,
reconnect fails and exception is raised.
(Ex: Mysql2::Error: Can't connect to local MySQL server through socket xxx)

But, ConnectionPool does not catch the exception, but leaks current
disconnected connection to @connection.
So, if database's temporary down happens several times and exceeds
the number of connection pool(5 by default), activerecord will be
no more available, even if database server is already recovered.

This patch fix it by catching exception and releasing connection.

Commit:
Update `rake counts` to count declarative tests in minitest

Commit:
Drop old test locations from `rake stats`

- test/functional
- test/unit

sgrif committed:
Merge pull request #18417 from yoshiokatsuneo/active_record_connection_pool_error_disconnect

ActiveRecord: release connection on reconnect failure.

rafaelfranca committed:
Merge pull request #18416 from caike/master

Add _mailer suffix to i18n path

jone committed:
Default headers, removed in controller actions, will not be reapplied to the test response.

senny committed:
Merge pull request #18423 from jone/remove-default-header

Default headers, removed in controller actions, will not be reapplied to the test response

Commit:
Merge pull request #18413 from brainopia/automatic_inverse_of_for_belongs_to

Correctly count methods when using declarative `#test` in minitest

rafaelfranca committed:
Merge pull request #18390 from tgxworld/parallel_bundle_install

Use parallel bundler install.

sgrif committed:
Properly persist `lock_version` as 0 if the DB has no default

The reason this bug occured is that we never actually check to see if
this column has changed from it's default, since it was never assigned
and is not mutable.

It appears I was wrong in b301c40224c6d15b539dbcc7485adb44d810f88c, with
my statement of "there is no longer a case where a given value would
differ from the default, but would not already be marked as changed."

However, I chose not to revert the deletion of
`initialize_internals_callback` from that commit, as I think a solution
closer to where the problem lies is less likely to get erroneously
removed. I'm not super happy with this solution, but it mirrors what is
being done in `_update_record`, and a fix for one should work for the
other.

I toyed with the idea of changing the definition of `changed?` on the
type to `changed_in_place?`. If we type cast the raw value, it'll break
a test about updating not modifying the lock column if nothing else was
changed. We could have the definition check if `raw_old_value` is `nil`,
but this feels fragile and less intention revealing. It would, however,
have the benefit of cleaning up old data that incorrectly persisted as
`nil`.

Fixes #18422

sgrif committed:
Properly copy nested bind values from subqueried relations

This is cropping up all over the place. After a brief dive, I'm really
not sure why we have `arel.bind_values` at all. A cursory grep didn't
reveal where they're actually being assigned (it's definitely in AR, not
in Arel). I'd like to dig further into it, as I'm fairly certain we
don't actually need it, we just need a way for the predicate builder to
communicate merged binds upstream.

Fixes #18414

arthurnn committed:
Use keyword args on committed! and rolledback!

As discussed before, those methods should receive a keyword args instead of just parameters

rafaelfranca committed:
Merge pull request #18425 from arthurnn/kargs_transaction

Use keyword args on committed! and rolledback!

arthurnn committed:
Copy records to parent transaction should happen on TransactionManager

It is up to the TransactionManager keep the state of current transaction, so after it commits it needs to copy any remaning record to the next current transaction

chancancode committed:
Merge pull request #18426 from arthurnn/txn_ref

Copy records to parent transaction should happen on TransactionManager

guilleiguaran committed:
Add SecureRandom.base58

robertomiranda committed:
Switch Secure Token generation to Base58

Update Secure Token Doc [ci skip]

remove require securerandom, core_ext/securerandom already do that ref 7e006057

guilleiguaran committed:
Add missing require

guilleiguaran committed:
Merge pull request #10690 from robertomiranda/remove_mass_assignment_security

Remove attributes_protected_by_default reference

guilleiguaran committed:
Merge pull request #18347 from robertomiranda/has_secure_token_base62

Switching SecureTokens to Base58

carlosantoniodasilva committed:
Remove support for the protected attributes gem

Related to #10690.

carlosantoniodasilva committed:
Remove note about supported plugins from the releasing docs

The last supported plugin on the list was protected_attributes, which is
not going to be supported anymore on Rails 5.

[ci skip]

senny committed:
formatting pass over CHANGELOGs. [ci skip]

koraktor committed:
Fix typo in PostresSQLAdapter's documentation

senny committed:
Merge pull request #18433 from koraktor/patch-1

Fix typo in PostresSQLAdapter's documentation [ci skip]

senny committed:
Merge branch 'Prefer-IO_NULL'. Closes #18381.

kuldeepaggarwal committed:
No need of requiring `rbconfig`, it is by-default loaded

deepj committed:
Better fork detection in ActiveSupport::Testing::Isolation

Process.respond_to?(:fork) returns false if fork is not available.
More on http://www.ruby-doc.org/core-2.0.0/Process.html#method-c-fork

eldano committed:
Use option url: false to allow entries without a link tag

xlgmokha committed:
allow '1' or true for acceptance validation.

sgrif committed:
Don't attempt to save dirty attributes which are not persistable

This sets a precident for how we handle `attribute` calls, which aren't
backed by a database column. We should not take this as a conscious
decision on how to handle them, and this can change when we make
`attribute` public if we have better ideas in the future.

As the composed attributes API gets fleshed out, I expect the
`persistable_attributes` method to change to
`@attributes.select(&:persistable).keys`, or some more performant
variant there-of. This can probably go away completely once we fully
move dirty checking into the attribute objects once it gets moved up to
Active Model.

Fixes #18407

sgrif committed:
Stop special casing null binary data in logging

There's very little value in logging "<NULL binary data>" instead of
just "nil". I'd like to remove the column from the equation entirely,
and this case is preventing us from doing so.

sgrif committed:
Stop passing a column to `quote` in `insert_fixture`

I'm planning on deprecating the column argument to mirror the
deprecation in [arel].

[arel]: https://github.com/rails/arel/commit/6160bfbda1d1781c3b08a33ec4955f170e95be11

sgrif committed:
Stop passing a column to `quote` when prepared statements are turned off

I'm planning on deprecating the column argument to mirror the
deprecation in [arel].

[arel]: https://github.com/rails/arel/commit/6160bfbda1d1781c3b08a33ec4955f170e95be11

sgrif committed:
Stop passing a column to `quote` in `Relation#to_sql`

I'm planning on deprecating the column argument to mirror the
deprecation in [arel].

[arel]: https://github.com/rails/arel/commit/6160bfbda1d1781c3b08a33ec4955f170e95be11

sgrif committed:
Stop passing a column to `quote` when executing from a statement cache

I'm planning on deprecating the column argument to mirror the
deprecation in [arel].

[arel]: https://github.com/rails/arel/commit/6160bfbda1d1781c3b08a33ec4955f170e95be11

sgrif committed:
Stop passing a column to `quote` when finding by AR models

I'm planning on deprecating the column argument to mirror the
deprecation in [arel].

[arel]: https://github.com/rails/arel/commit/6160bfbda1d1781c3b08a33ec4955f170e95be11

sgrif committed:
Stop passing a column to `quote` in tests

I'm planning on deprecating the column argument to mirror the
deprecation in [arel].

[arel]: https://github.com/rails/arel/commit/6160bfbda1d1781c3b08a33ec4955f170e95be11

robertomiranda committed:
Fix Typo SecureToken [ci skip]

guilleiguaran committed:
Merge pull request #18440 from robertomiranda/patch-3

Fix Typo SecureToken [ci skip]

sgrif committed:
Deprecate passing a column to `quote`

It's only used to grab the type for type casting purposes, and we would
like to remove the type from the columns entirely.

prathamesh-sonpatki committed:
Run SQL only if attribute changed for update_attribute method

 - This is based on https://github.com/rails/rails/issues/18400 but
   tackling same issue with update_attribute method instead of update method.

y-yagi committed:
remove autoload HideActions, no longer exist

rafaelfranca committed:
Merge pull request #18445 from y-yagi/remove_hide_actions

remove autoload HideActions, no longer exist

rafaelfranca committed:
Merge pull request #18435 from kuldeepaggarwal/f-remove-unnecessary-require

No need of requiring `rbconfig`, it is by-default loaded

tonytonyjan committed:
fix dependency

robertomiranda committed:
Add Secure Token Generator

jlduran committed:
Re-add section in SQL Caching

This section was accidentally removed in 4b5e424441149cf0f61a8182f4a92999577ce59a, and partially re-added in dc14b893e5904acb37ac327478f3880f33094b7a.

[ci skip]

claudiob committed:
Fix assertion that was never run

In order to run whether the `welcome` method of the ActionMailer::Base
subclass raises an error, `message` must be called, otherwise the method
is not executed at all.

You could just replace with `def welcome; raise StandardError; end` and you
would still see a passing test.

This commit fixes the test so the assertion is actually executed, just like
any other tests in the file, where `.message` is called.

rafaelfranca committed:
Merge pull request #18452 from claudiob/fix-broken-assertion

Fix assertion that was never run

rafaelfranca committed:
Merge pull request #18451 from jlduran/guides-sql-caching

Re-add section in SQL Caching [ci skip]

guilleiguaran committed:
Merge pull request #18448 from robertomiranda/has_secure_token_generator

Add has secure token to migration generator

claudiob committed:
Add test for AM::Validation::Callbacks with :on

`before_validation` and `after_validation` from
ActiveModel::Validation::Callbacks accept an optional `:on` parameter
that was not previously documented or tested. For instance given

    before_validation :do_something, on: :create

then `object.valid?(:create)` will invoke `:do_something` while
`object.valid?` or `object.valid?(:anything_else)` will not.

claudiob committed:
Remove unused "deprecated_reload" method

The method was introduced in https://github.com/rails/rails/commit/66d0a0153578ce760d822580c5b8c0b726042ac2#diff-8cec05860729a3851ceb756f4dd90370R49
for the "reset_changes is deprecated" test, but this test was successively
removed in https://github.com/rails/rails/commit/37175a24bd508e2983247ec5d011d57df836c743

claudiob committed:
Remove unused AV fixtures from AP tests

When `render` was moved from ActionPack to ActionView in acc8e259,
some fixtures required by the tests were duplicated, but they are
actually only required by ActionView tests.

To give one example, `double_render` is already defined [in the AV tests](https://github.com/rails/rails/blob/72139d8d310d896db78eaec98582c7a638135102/actionview/test/actionpack/controller/render_test.rb#L407)
and is never used in the ActionPack tests.

carlosantoniodasilva committed:
Merge pull request #18455 from claudiob/remove-unused-deprecated_reload-method

Remove unused "deprecated_reload" method

carlosantoniodasilva committed:
Merge pull request #18454 from claudiob/test-on-option-for-amv-callbacks

Add test for AM::Validation::Callbacks with :on

carlosantoniodasilva committed:
Merge pull request #18456 from claudiob/remove-unused-fixtures-to-render

Remove unused AV fixtures from AP tests

carlosantoniodasilva committed:
Merge pull request #18436 from deepj/fork-detection

Better fork detection in ActiveSupport::Testing::Isolation

Commit:
after_commit runs after transactions with non-joinable parents

after_commit callbacks run after committing a transaction whose parent
is not `joinable?`: un-nested transactions, transactions within test
cases, and transactions in `console --sandbox`.

mechanicles committed:
Change 'a' to 'an' for 'XML' word [ci skip]

imogenkinsman committed:
minor grammatical fix in documentation

matthewd committed:
Merge pull request #18461 from zetsubo/patch-1

minor grammatical fix in documentation

fxn committed:
Merge pull request #18460 from mechanicles/a-to-an-for-xml

Change 'a' to 'an' for 'XML' word [ci skip]

davydovanton committed:
Fix error messages scope [skip ci]

seuros committed:
Merge pull request #18466 from davydovanton/error_message_doc

Fix description for ActiveModel::Errors#generate_message [skip ci]

spastorino committed:
Relax Rack dependency

Rack is very carefully released, we should be able to upgrade minor
versions without much effort. We are a bunch of Rails core who are also
Rack core members so there won't be any issue with that. And in case
there's something wrong, we should fix on both sides.
Even though, doesn't seem like we will have a 1.7 version, this will be
useful as an example for when we go with Rack 2.0. We should ~> 2.0.

dmasur committed:
Removing :en in favor of I18n.default_locale in duration#inspect

Hi there,

i have an app without english as available locale. So i got an error when we try to inspect something like 1.day. This is done automatically when we use the dalli cache.

I would like to change the :en to ::I18n.default_locale to be sure that this is always constant and is an available locale.

Tests are all green with this change.

Calculating -------------------------------------
:locale => :en                              2.024k i/100ms
:locale => ::I18n.default_locale   2.236k i/100ms
-------------------------------------------------
:locale => :en                                  25.758k (±26.3%) i/s -    117.392k
:locale => ::I18n.default_locale       26.311k (±18.1%) i/s -    127.452k

senny committed:
Merge pull request #18462 from dmasur/patch-1

Removing :en in favor of default locale for duration#inspect

sgrif committed:
Merge pull request #18439 from mokhan/validates-acceptance-of-array

allow '1' or true for acceptance validation.

ahmad-alkheat committed:
Updated active_record_querying.md

Added the SQL equivalent of the find_by method

Update active_record_querying.md

Update active_record_querying.md

kamipo committed:
Remove unused accessor

Commit:
Clean up grammar in scaffold index view, by removing the word 'Listing' from list title.

yoshiokatsuneo committed:
ActiveRecord: On reconnection failure, release only failed connetion.

On reconnection failure, all the connection was released.
But, it is better to release only failed connection.
This patch changes not to release all the connection but release
only failed connection.

mechanicles committed:
Change a to an for HTML word [ci skip]

sevenseacat committed:
Digesting is now enabled by default in development

claudiob committed:
Merge pull request #18479 from mechanicles/change-a-to-an-for-html

Change a to an for HTML word [ci skip]

claudiob committed:
Merge pull request #18480 from karpah/patch-1

Update asset pipeline guide

rono23 committed:
Fix name_for_action in routing

rafaelfranca committed:
Merge pull request #18471 from ahmad-alkheat/master

Added the SQL equivalent of the find_by method. [ci skip]

Envek committed:
Add missing time zone definitions for Russian Federation

Also sync Russian Federation time zones with `zone.tab` file from tzdata version 2014j.

Commit:
Set all asset options together

Commit:
Show how to change the queuing backend for ActiveJob in production

Commit:
Stop promoting rack-cache usage at the moment (not so common or important to deserve that billing)

sgrif committed:
Add the `method_source` gem to the default Gemfile

Fixes #18473

Commit:
Merge pull request #18476 from Alamoz/scaffold_index_view_grammar

Clean up grammar in scaffold index view, by removing the word 'Listing' ...

andreynering committed:
Guides: Removing reference to blog that is not updated anymore [ci skip]

seuros committed:
Merge pull request #18490 from andreynering/security-guides-remove-resource

Guides: Removing reference to blog that is not updated anymore [ci skip]

Commit:
Fix form_for guide binding a form to an object.

The HTML generated using
    url: { action: :create}
will not generate the form action "/articles/create", it should generate
the form action "/articles" for a new object.

Commit:
Use consistent anchor text throughout the guide.

Change all of anchor text in the nav links to `Back` and `Edit`
Use `Edit | Back` on pages that use both links to conform to what
scaffolding generates.

Use consistent anchor text throughout the guide.

Change all of anchor text in the nav links to `Back` and `Edit`
Use `Edit | Back` on pages that use both links to conform to what
scaffolding generates.

pixeltrix committed:
Merge pull request #18489 from Envek/ru_timezones

Add missing time zone definitions for Russian Federation

seuros committed:
Merge pull request #18491 from dreeve/form-for-guide-fix-generated-action

Fix form_for guide binding a form to an object. [ci skip]

sgrif committed:
Merge pull request #18478 from yoshiokatsuneo/active_record_connection_pool_error_disconnect

ActiveRecord: On reconnection failure, release only failed connetion.

nippysaurus committed:
improve grammar

y-yagi committed:
fix link in autoloading guide [ci skip]

vipulnsward committed:
Fixed spacing issues on code comments in ActiveJob Basics guide.

Before: http://take.ms/z39h2
After: http://take.ms/Mo0G3

vipulnsward committed:
- Changed `IN` to `ON` in markdown renderer condition
- Changed `IN` to `ON` in all note sentences in guides.

vipulnsward committed:
Extract `Date` header to string constant in Http Response, similar to other headers

senny committed:
Merge pull request #18502 from y-yagi/fix_autoloading_guide

fix link in autoloading guide [ci skip]

senny committed:
Merge pull request #18504 from vipulnsward/extract-date

Extract `Date` header to string constant in Http Response, similar to other headers

senny committed:
Merge pull request #18500 from nippysaurus/typo_fix

Grammar: "By default, Rails loads generators from your load path." [ci skip]

amatsuda committed:
require 'rails' if not include_all_railties, just as 'rails/all' does

fxn committed:
Merge pull request #18503 from vipulnsward/guides-in-on

Changed `IN` to `ON` in markdown renderer condition

prathamesh-sonpatki committed:
Fix typo in ActionView CHANGELOG [ci skip]

robin850 committed:
Merge pull request #18509 from prathamesh-sonpatki/cg-typo

 Fix typo in ActionView CHANGELOG [ci skip]

prathamesh-sonpatki committed:
Changed "in GitHub" to "on GitHub" [ci skip]

 - Followup of https://github.com/rails/rails/pull/18503.

fxn committed:
Merge pull request #18510 from prathamesh-sonpatki/some-more-in-on-github

 Changed "in GitHub" to "on GitHub" [ci skip]

andrewvida committed:
Correct views mail method sends in API docs

rafaelfranca committed:
Merge pull request #18521 from andrewvida/master

Correct the views that the mail method sends in API docs.

sgrif committed:
Don't default to YAML dumping when quoting values

This behavior exists only to support fixtures, so we should handle it
there. Leaving it in `#quote` can cause very subtle bugs to slip
through, by things appearing to work when they should be blowing up
loudly, such as #18385.

sgrif committed:
Go through normal where logic in destroy

Building the Arel AST, and manipulating the relation manually like this
is prone to errors and breakage as implementation details change from
underneath it.

sgrif committed:
Go through normal where logic in destroy with locking

Building the Arel AST, and manipulating the relation manually like this
is prone to errors and breakage as implementation details change from
underneath it.

sgrif committed:
Don't use the `_before_type_cast` version of attributes in the form

We should never be ignoring valuable information that the types may need
to give us. The reason that it originally used `_before_type_cast` is
unclear, but appears to date back long enough that the reasons may not
be relevant today. There is only one test that asserts that it uses the
before type cast version, but it gives no context as to why and uses a
mock which does not simulate the real world.

Fixes #18523.

sgrif committed:
Revert "Don't use the `_before_type_cast` version of attributes in the form"

This reverts commit 787e22bb491bd8c36db1e9734261c4ce02c5c5fd.

sgrif committed:
Add tests with an explanation of why we're using `_before_type_cast`

If you entered incorrect input into a form (particularly with a
numericality validation), we should not replace what you typed in with
`0`, since clicking submit a second time would essentially be the same
as erroneously accepting the junk input as 0.

sgrif committed:
Only use the `_before_type_cast` in the form when from user input

While we don't want to change the form input when validations fail,
blindly using `_before_type_cast` will cause the input to display the
wrong data for any type which does additional work on database values.

vijaydev committed:
Merge branch 'master' of github.com:rails/docrails

Commit:
Add `ActionController::Metal#set_request!`

Add `ActionController::Metal#set_request!` to set a request
on controller instance without calling dispatch.

Commit:
Support `:assigns` option when rendering with controllers/mailers.

Commit:
Fix Typo SecureToken for schema sample [ci skip]

sgrif committed:
Merge pull request #15726 from sgrif/sg-time-zone-aware-times

Time columns should support time zone aware attributes

Commit:
Merge pull request #18537 from jrnk/patch-1

Fix Typo SecureToken for schema sample [ci skip]

Commit:
[ci skip] fix typo sill -> still

seuros committed:
Merge pull request #18539 from aditya-kapoor/correct-typo

[ci skip] fix typo sill -> still

andreynering committed:
Adding tip about jQuery UJS [ci skip]

Reference #18533

seuros committed:
Merge pull request #18541 from andreynering/getting-started-javascript

Adding tip about jQuery UJS [ci skip]

henrik committed:
DRY default limit in ActiveRecord::Type::Integer

henrik committed:
Remove incorrect comment in ActiveRecord::Type::Value

Says it's only used for the schema, but they are in fact used for
other things. Integer verifies against the limit during casting,
and Decimal uses precision during casting. It may be true that
scale is only used for the schema.

Commit:
Add ActionController#build_with_env

To have an easier way to setup a controller
instance with custom environment

rafaelfranca committed:
Merge pull request #18544 from henrik/ar_type_value_comment

Tiny: Remove incorrect comment in ActiveRecord::Type::Value

senny committed:
Merge pull request #18543 from henrik/integer_limit_or_default

Tiny: DRY default limit in ActiveRecord::Type::Integer

senny committed:
docs, update "How to run Active Record tests". [ci skip]

mechanicles committed:
A little performance improvement using 'take' method over 'first'

sgrif committed:
Specify correct version in the changelog [ci skip]

spastorino committed:
Merge pull request #18437 from eldano/feed_entry_no_link_option

Use option url: false to allow entries without a link tag

Commit:
Fix assignment for frozen value in HWIA

jfine committed:
Underscore typo [ci skip]

rafaelfranca committed:
Merge pull request #18556 from jfine/patch-1

Underscore typo [ci skip]

tjschuck committed:
Fix code formatting in sanitize docs

pixeltrix committed:
Merge pull request #18551 from aditya-kapoor/fix-frozen-hwia

Fix assignment for frozen value in HWIA

tjschuck committed:
Rewrite SanitizeHelper docs

robertomiranda committed:
Update Molde Generator Usage

Including has_secure_token generator

ogennadi committed:
Typo fix

Commit:
Merge pull request #18559 from ogennadi/patch-1

Typo fix

ogennadi committed:
Fixed extraneous quoting

seuros committed:
Merge pull request #18560 from ogennadi/patch-2

Fixed extraneous quoting

vipulnsward committed:
- Moved hwia frozen value assignment test to hash_ext_test similar to other tests
- Fixed the wrong use of with_indifferent_access on hash in the test which failed for isolated tests
- Renamed to appropriately specify what the test does

spastorino committed:
Merge pull request #18562 from vipulnsward/fix-broken-hwai-frozen-tet

Fix hwia isolated tests

spastorino committed:
Merge pull request #18554 from mechanicles/use-take-method

A little performance improvement using 'take' method over 'first'

senny committed:
Merge pull request #18558 from robertomiranda/patch-3 [ci skip]

Update Molde Generator Usage

senny committed:
Merge pull request #18493 from twoller/master

Consistent anchor text in nav links, new image [ci skip]

yui-knk committed:
[ci skip] Change to the passive voice

seuros committed:
Merge pull request #18564 from yui-knk/fix/autoload2

[ci skip] Change to the passive voice

yuki24 committed:
Fix a false assertion

#assert was used when it should be assert_equal.

carlosantoniodasilva committed:
Merge pull request #18566 from yuki24/fix-assertion-in-ar-case-date-time

Fix a false assertion

prathamesh-sonpatki committed:
Fixed test for assert_welcome

 - Test was wrongly using assert with assignment statement.

vipulnsward committed:
Removed documentation of deprecated removed methods [ci skip]

Commit:
Merge pull request #18568 from vipulnsward/removed-deprecated-methods-from-docs

Removed documentation of deprecated removed methods

senny committed:
bring back unintentionally removed blank line [ci skip]

Thanks @carlosantoniodasilva :yellow_heart:

schneems committed:
Merge pull request #18567 from prathamesh-sonpatki/fix-assert-welcome-test

 Fixed test for assert_welcome

yuki24 committed:
Eliminate a ruby warning in a test in Action View

This eliminates the warning below:

  actionview/test/template/form_helper_test.rb:913: warning: method redefined; discarding old id_came_from_user?
  actionview/test/template/form_helper_test.rb:104: warning: previous definition of id_came_from_user? was here

senny committed:
Merge pull request #18569 from yuki24/eliminate-warning-in-action-view-test

Eliminate a ruby warning in a test in Action View

Commit:
Add ActionController::Renderer

Render arbitrary templates outside of controller actions

yui-knk committed:
[ci skip] fix positions of parentheses

yui-knk committed:
[ci skip] Move a introduction of `Module.nesting`

Move a introduction of `Module.nesting` to the beginning of sentence.
This change will help reader to try to dump `nesting`.

Commit:
Merge pull request #18575 from yui-knk/fix/autoload

[ci skip] fix positions of parentheses

prathamesh-sonpatki committed:
nodoc ActionView::ModelNaming module

- Its only used by ActionView internals and not supposed to be used
  through public API.

y-yagi committed:
remove reference to `Logger` core extensions from guide [ci skip]

guilleiguaran committed:
Merge pull request #18577 from prathamesh-sonpatki/nodoc-av-modelnaming

nodoc ActionView::ModelNaming module

seuros committed:
Merge pull request #18578 from y-yagi/remove_logger_extensions

remove reference to `Logger` core extensions from guide [ci skip]

prathamesh-sonpatki committed:
Use existing constant for joining instead of '_'

spastorino committed:
Merge pull request #18579 from prathamesh-sonpatki/use-existing-constant-for-join

Use existing constant for joining instead of '_'

spastorino committed:
Merge pull request #18576 from yui-knk/fix/autoload4

[ci skip] Move a introduction of `Module.nesting`

sgrif committed:
Include stackprof on ruby 2.2 as well as 2.1

sgrif committed:
Use bundler 1.7 on travis

sgrif committed:
Don't calculate in-place changes on attribute assignment

When an attribute is assigned, we determine if it was already marked as
changed so we can determine if we need to clear the changes, or mark it
as changed. Since this only affects the `attributes_changed_by_setter`
hash, in-place changes are irrelevant to this process. Since calculating
in-place changes can be expensive, we can just skip it here.

I also added a test for the only edge case I could think of that would
be affected by this change.

kamipo committed:
Should escape regexp wildcard character `.`

`.` is regexp meta character. It should be escape for `assert_match`
correctly.

fxn committed:
Merge pull request #18586 from kamipo/should_escape_regexp_wildcard_character

Should escape regexp wildcard character `.`

spastorino committed:
Merge pull request #18501 from prathamesh-sonpatki/nosql

 Run SQL only if attribute changed for update_attribute method

spastorino committed:
Merge pull request #18591 from kamipo/remove_unused_accessor

Remove unused accessor

spastorino committed:
Merge pull request #18380 from nobu/Use-EXEEXT

Use EXEEXT

spastorino committed:
Merge pull request #18483 from rono23/fix-name_for_action-in-routing

Fix name_for_action in routing

prathamesh-sonpatki committed:
Fix wording in AR CHANGELOG about time columns change [ci skip]

prathamesh-sonpatki committed:
Use force: true options for creating tables in bug templates

 - Generally we have to run the bug templates multiple times to get them
   right and it always complains because the posts and comments tables
   already exist due to earlier runs.
 - Using force: true will eliminate this issue.

y-yagi committed:
fix typo in number_field_tag documentation [ci skip]

seuros committed:
Merge pull request #18600 from y-yagi/fix_number_field_tag_example

fix typo in number_field_tag documentation [ci skip]

senny committed:
Merge pull request #18594 from prathamesh-sonpatki/use-force-true-in-templates

 Use force: true options for creating tables in bug templates

senny committed:
Merge pull request #18593 from prathamesh-sonpatki/ar-changelog

 Fix wording in AR CHANGELOG about time columns change [ci skip]

rafaelfranca committed:
Merge pull request #18450 from tonytonyjan/patch-1

fix dependency

sgrif committed:
Don't mutate bind values in `Relation`

In order to better facilitate refactoring, most places that mutated
`bind_values` have already been removed. One last spot snuck through.
Since we're no longer mutating the array, it also does not need to be
duped in `initialize_copy`.

sgrif committed:
Whether a column exists or not doesn't affect whether we can use binds

Looking through the blame, this logic used to be when we actually
created the bind tuple. My guess is that `nil` couldn't be handled there
at that time. It can, now.

sgrif committed:
Move `create_binds` over to the `PredicateBuilder`

I'm looking to introduce a `WhereClause` class to handle most of this
logic, and this method will eventually move over to there. However, this
intermediate refactoring should make that easier to do.

sgrif committed:
Fix bind value copying from subqueried relations

With the old implementation, the bind values were created, and then we
search the attributes for `Relation` objects, and merge them. This
completely ignores the order that the actual `where` clause will use. If
all non-relation where parameters are before the relations, it will
work. However, if we query on both a relation and a value, with the
value coming second, it breaks. The order of the hash should not affect
the final query (especially since hashes being ordered is an
implementation detail)

senny committed:
Merge pull request #18597 from kamipo/add-if-exists-to-drop-table

Add an `:if_exists` option to `drop_table`

senny committed:
tests, use `drop_table if_exists: true` in our test suite.

jeremy committed:
Merge pull request #18458 from brainopia/fix_after_commit_for_fixtures

Support after_commit callbacks in transactional fixtures

arthurnn committed:
TransactionManager should call rollback records

rafaelfranca committed:
Merge pull request #18614 from arthurnn/rollback_records_parity

TransactionManager should call rollback records

tjschuck committed:
Single instead of double quotes	

[ci skip]

sgrif committed:
Introduce `ActiveRecord::Base#accessed_fields`

This method can be used to see all of the fields on a model which have
been read. This can be useful during development mode to quickly find
out which fields need to be selected. For performance critical pages, if
you are not using all of the fields of a database, an easy performance
win is only selecting the fields which you need. By calling this method
at the end of a controller action, it's easy to determine which fields
need to be selected.

While writing this, I also noticed a place for an easy performance win
internally which I had been wanting to introduce. You cannot mutate a
field which you have not read. Therefore, we can skip the calculation of
in place changes if we have never read from the field. This can
significantly speed up methods like `#changed?` if any of the fields
have an expensive mutable type (like `serialize`)

```
Calculating -------------------------------------
 #changed? with serialized column (before)
                       391.000  i/100ms
 #changed? with serialized column (after)
                         1.514k i/100ms
-------------------------------------------------
 #changed? with serialized column (before)
                          4.243k (± 3.7%) i/s -     21.505k
 #changed? with serialized column (after)
                         16.789k (± 3.2%) i/s -     84.784k
```

rafaelfranca committed:
Merge pull request #18557 from tjschuck/sanitize_doc_fixes

SanitizeHelper documentation formatting fixes + rewrite

yahonda committed:
Replace `if exists` with `table_exists?` and drop table statement with `drop_table`
since 'drop table if exists' statement does not always work with some databases such as Oracle.
also Oracle drop table statement will not drop sequence objects.

morgoth committed:
Add missing AS core extension dependency

mechanicles committed:
Change 'a' to 'an' for 'attribute' word [ci skip]

fxn committed:
Merge pull request #18620 from mechanicles/change-a-to-an

Change 'a' to 'an' for 'attribute' word [ci skip]

morgoth committed:
Add missing options to Error Message Interpolation guide [CI skip]

carlosantoniodasilva committed:
Merge pull request #18621 from morgoth/error-message-on-validation-guides

Add missing options to Error Message Interpolation guide [CI skip]

carlosantoniodasilva committed:
Merge pull request #18619 from morgoth/add-missing-dependency-to-activemodel-naming

Add missing AS core extension dependency

Commit:
Travis: Test on jruby-head only for now

Arel head does not support 1.9 anymore.

rafaelfranca committed:
Merge pull request #18626 from splattael/jruby-head

Travis: Test on jruby-head only for now

rafaelfranca committed:
Fix test case class in the testing guide [ci skip]

rafaelfranca committed:
Merge pull request #18628 from yahonda/if_exists_testings

Replace `if exists` with `table_exists?` and drop table with `drop_table`

rafaelfranca committed:
Merge pull request #18322 from morgoth/add-error-codes

Add ActiveModel::Errors#codes

sgrif committed:
Don't error when invalid json is assigned to a JSON column

Keeping with our behavior elsewhere in the system, invalid input is
assumed to be `nil`.

Fixes #18629.

Commit:
Add ApplicationController.renderer initializer [ci skip]

Commit:
Add ActionController::Base.render

BenMorganIO committed:
use two spaces instead of one for include root in json

josh committed:
Remove config.assets

Provided by sprockets-rails plugin

diogobenica committed:
Add :formats to options list of render documentation.

arthurnn committed:
Merge pull request #18637 from diogobenica/master

Add :formats to options list of render documentation.

victorteokw committed:
Fixed undefined method error when doing authentication.

egilburg commented on actionpack/lib/action_controller/metal/http_authentication.rb:
```diff
@@ -106,11 +106,11 @@ def decode_credentials(request)
 
       def auth_scheme(request)
-        request.authorization.split(' ', 2).first
+        (request.authorization || '').split(' ', 2).first
```
Can do `request.authorization.to_s.split('  ', 2).first`

`.to_s` will not make a new object if orginal is already a string, so shouldn't be a performance issue.

cheunghy replied:
Thanks.

Will it be a performance issue to evaluate ``request.authorization`` twice?
``` ruby
request.authorization.split(' ', 2).first if request.authorization
```

senny committed:
Merge pull request #18635 from BenMorganIO/use-two-spaces-instead-of-one-for-wrap-params

use two spaces instead of one for include root in json

senny commented:
@cheunghy could you add a test-case for this?

Commit:
Disable builds AR-JDBC against master

Currently, bundling `activerecord-jdbc-adapter`'s master branch seems to
be broken. See jruby/activerecord-jdbc-adapter#614

This commit partially reverts #12107

y-yagi committed:
fix regexp for validate an integer in guides [ci skip]

rafaelfranca committed:
Merge pull request #18546 from brainopia/action_view_render

A shortcut to setup controller environment

rafaelfranca committed:
Merge pull request #18636 from josh/railties-remove-config-assets

Remove config.assets

seuros committed:
Merge pull request #18642 from y-yagi/active_record_validations_guide

fix regexp for validate an integer in guides [ci skip]

rafaelfranca committed:
Merge pull request #18641 from splattael/disable-ar-jdbc-master

Disable builds AR-JDBC against master

kaspth committed:
Speed up `normalize_keys` by removing dup step.

Previously env was duplicated and then had it's keys mutated. This iterates through
the hash twice.
Using `transform_keys`, duplication and key mutation is a single iteration.

`convert_symbols` was renamed to `http_header_format`.

rwarbelow committed:
fix wording of CSS manifest instructions

Commit:
Merge pull request #18644 from rwarbelow/fix-wording-of-css-manifest

fix wording of CSS manifest instructions [ci skip]

rafaelfranca committed:
Merge pull request #18643 from kaspth/renderer-normalize_keys-speedup

Speed up ActionController::Renderer `normalize_keys` by ~28%.

victorteokw committed:
Add test for HTTP basic authentication when no credential is given.

cheunghy commented:
@senny Test added.

mechanicles committed:
Use 'public_send' over the 'send' method for object's properties.

fxn committed:
constants guide: make assumption explicit in an example [closes #18465] [ci skip]

bogdan committed:
Extracted `ActiveRecord::AttributeAssignment` to `ActiveModel::AttributesAssignment`

Allows to use it for any object as an includable module.

rafaelfranca commented:
Could you open the pull request against master branch? We may decide to not backport to 4-2-stable.

rafaelfranca committed:
Merge pull request #18651 from mechanicles/use-public-send

Use 'public_send' over the 'send' method for object's properties and public methods.

sgrif committed:
Don't remove join dependencies in `Relation#exists?`

Fixes #18632

sgrif committed:
Errors raised in `type_cast_for_database` no longer raise on assignment

Fixes #18580.

sgrif committed:
Move integer range validation to never raise on assignment

Given that this was originally added to normalize an error that would
have otherwise come from the database (inconsistently), it's more
natural for us to raise in `type_cast_for_database`, rather than
`type_cast_from_user`. This way, things like numericality validators can
handle it instead if the user chooses to do so. It also fixes an issue
where assigning an out of range value would make it impossible to assign
a new value later.

This fixes several vague issues, none of which were ever directly
reported, so I have no issue number to give. Places it was mentioned
which I can remember:

- https://github.com/thoughtbot/shoulda-matchers/blob/9ba21381d7caf045053a81f32df7de2f49687820/lib/shoulda/matchers/active_model/allow_value_matcher.rb#L261-L263
- https://github.com/rails/rails/issues/18653#issuecomment-71197026

sgrif committed:
✂️ and 💅 for #10776

Minor style changes across the board. Changed an alias to an explicit
method declaration, since the alias will not be documented otherwise.

sgrif committed:
Merge pull request #10776 from bogdan/assign-attributes

Extracted attributes assingment from ActiveRecord to ActiveModel

cheunghy commented:
https://github.com/rails/rails/pull/18666

sgrif commented:
There is no need to open pull requests against 4-2-stable, we will backport the merge commit from a PR made against master. This also includes the entire master branch.

cheunghy commented:
Thanks!
Yes, I have already move it to https://github.com/rails/rails/pull/18666.
